### DISCUSSION TITLE
Filter a structured collection?

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#6116] [General]
Hi

I'm using an ordered collection as navigation `nav:collection:products`, the collection has a `taxonomy` called divisions.

I'd like to filter `nav:collection:products` by a taxonomy term. 

I can only get the filter to work if I use `collection:products`, I can't get it to work with `nav:collection:products`

### collection:products
When I use  `collection:products`, the `taxonomy:divisions` filter works as expected, the entires are filtered by a taxonomy term, **but** It ignores the children - because it's not a `nav`
```
 {{ collection:products taxonomy:divisions="{slug}"  }}
            <ul>
                <li>{{ title }} - {{ divisions.title | wrap:span.font-semibold }}</li>
                {{ if children }}
                    <ul class="ml-4">
                        {{ children }}
                            <li>
                                {{ title }}
                                {{ if children }}
                                    <ul class="ml-4">{{ *recursive children* }}</ul>
                                {{ /if }}
                            </li>
                        {{ /children }}
                    </ul>
                {{ /if }}
            </ul>
{{ /collection:products }}
```
### nav:collection:products
If I add the `nav:` part, the `taxonomy:divisions="{slug}" ` is ignored - so all the entires are displayed, and the children are shown.

```
{{ nav:collection:products taxonomy:divisions="{slug}"  }}
            <ul>
                <li>{{ title }} - {{ divisions.title | wrap:span.font-semibold }}</li>
                {{ if children }}
                    <ul class="ml-4">
                        {{ children }}
                            <li>
                                {{ title }}
                                {{ if children }}
                                    <ul class="ml-4">{{ *recursive children* }}</ul>
                                {{ /if }}
                            </li>
                        {{ /children }}
                    </ul>
                {{ /if }}
            </ul>
        {{ /nav:collection:products }}
```

I'd like to be able to filter `nav:collection:products` by `taxonomy:divisions="{slug}"`

Is there a way to get this to work?

```
Statamic 3.3.12 Pro
Laravel 9.12.2
PHP 8.0.8
Parser: runtime
```

**Comment by afonic** [comment]
I don't think the `nav` tag is meant to support all the `collection` tag methods so this is more of a feature request.

For now, you should probably export the markup you need to a partial and try to make it work using the collection tag.

**Reply by stephenmeehanuk** [reply]
Thanks, but I'm not sure how I'd go about making it work, if Statamic doesn't support filtering the `nav` tag like this.

**Reply by edalzell** [reply]
Hey good news, it looks like the `nav` tag DOES support filters, just not the special taxonomy one, which means this is doable I think!

Try: `{{ nav:collection:products :divisions:is="slug" }}`

**Reply by stephenmeehanuk** [reply]
Thanks, unfortunately `{{ nav:collection:products :divisions:is="slug"  }}` causes a server error?

```
local.ERROR: strtolower(): Argument #1 ($string) must be of type string, array given {"exception":"[object] (TypeError(code: 0): strtolower(): Argument #1 ($string) must be of type string...
```
Would this be a bug?

**Reply by stephenmeehanuk** [reply]
Is there anything else I could try?

**Reply by afonic** [reply]
Custom modifier could maybe work?

**Reply by edalzell** [reply]
`{{ nav:collection:products divisions:is="{slug}" }}`?

**Reply by stephenmeehanuk** [reply]
I should have mentioned, I tried that and it does the same thing. Errors the page. 

Is either one of those supposed to  work? If so, should I post it as a bug?

**Reply by edalzell** [reply]
if you `{{ slug | dump }}` what do you get?

**Reply by stephenmeehanuk** [reply]
It outputs the correct `slug`. This matches the name of the `divisions` taxonomy `term` I'm trying to filter `nav:collection:products` on.

.....

_I managed to figure out what was going on._

```
Argument #1 ($string) must be of type string, array given
```

The answer was right in front of me. I double checked the divisions terms data

At some point the taxonomy allowed multiple terms (array), but had been restricted to one term in the Blueprint. But for some of the entires, the data was still formatted as an array.

Thanks for your help on this.

**Comment by afonic** [comment]
I guess the entries are so many that making a nav for this collection is out of the question right?

Indeed, without access to the structure you this would be hard to do. How about trying a custom modifier for filtering the results of the nav tag?
